vendor/zkllvm-metacraft-circuits/src/utils/picosha2.h

```diff
@@ -158,39 +186,44 @@ namespace picosha2 {
 
         void init() {
             buffer_size_ = 0;
-            std::fill(data_length_digits_, data_length_digits_ + 4, word_t(0));
-            std::copy(detail::initial_message_digest, detail::initial_message_digest + 8, h_);
+            fill(data_length_digits_, data_length_digits_ + 4, word_t(0));
+            copy(detail::initial_message_digest, detail::initial_message_digest + 8, h_);
         }
 
         template<typename RaIter>
         void process(RaIter first, RaIter last) {
             add_to_data_length(static_cast<word_t>(std::distance(first, last)));
             assert_true(buffer_size_ + (last - first) < PICOSHA2_BUFFER_SIZE_FOR_INPUT_ITERATOR);
-            std::copy(first, last, buffer_.begin() + buffer_size_);
+            copy(first, last, buffer_.begin() + buffer_size_);
             buffer_size_ += last - first;
-            std::size_t i = 0;
-            for (; i + 64 <= buffer_size_; i += 64) {
-                detail::hash256_block(h_, buffer_.begin() + i, buffer_.begin() + i + 64);
+            std::size_t processed = 0;
+            for (size_t i = 0; i + 64 <= PICOSHA2_BUFFER_SIZE_FOR_INPUT_ITERATOR; i += 64) {
```

Comment:
get rid of `PICOSHA2_BUFFER_SIZE_FOR_INPUT_ITERATOR` in the future. It is the cause of many wasted iterations

Reply:
The idea is that this constant look arbitrary. I think it would be better to supply the buffer size as a template parameter to the function. If you think there will be no need for this with the crypto3 lib, then pass on this comment